hass-addon-sunsynk-dev/run.py

```diff
@@ -179,10 +202,23 @@ def startup() -> None:
 def setup_sensors() -> None:
     """Setup the sensors."""
     sens = {}
+    sens_dependencies: Dict[str, List[Sensor]] = defaultdict(list)
     startup_sens = {SERIAL.id, RATED_POWER.id}
+    filters: Dict[str, Filter] = {}
 
     msg: Dict[str, List[str]] = defaultdict(list)
 
+    def enqueue_hass_discovery_info_update(sen: Sensor, deps: List[Sensor]):
+        if not DEVICE:
```

Comment:
The `on_change` callables fire even on startup when sensor values are `None` and are read for the first time. This check prevents unnecessary queuing of discovery info updates.